File: NotificationService/Application/CQRS/DTO/SendNotificationCommand.cs
using System;
using System.Collections.Generic;
using MediatR;

namespace NotificationService.Application.CQRS.DTO
{
    public record SendNotificationCommand(
        string SagaId,
        string TransactionId, // ID of the overall transaction
        string UserId, // The user to notify
        string RecipientAddress, // e.g., email address, phone number, device token
        string MessageType, // e.g., "EMAIL", "SMS", "PUSH"
        string Subject, // Title for email/push
        string Body, // Content of the notification
        Dictionary<string, string> TemplateModel // For personalized notifications
    ) : IRequest<NotificationSentReply>;
} 

File: NotificationService/java-listener/src/main/java/com/webbanking/notificationservice/dto/SendNotificationCommand.java
package com.webbanking.notificationservice.dto;

import java.util.Map;

/**
 * Command object for requesting notification service to send a notification
 */
public record SendNotificationCommand(
        String sagaId,
        String transactionId, // ID of the overall fund transfer transaction
        String userId, // The user to notify
        String recipientAddress, // e.g., email address, phone number, device token
        String messageType, // e.g., "EMAIL", "SMS", "PUSH"
        String subject, // Title for email/push
        String body, // Content of the notification
        Map<String, String> templateModel // For personalized notifications
) {
} 